### PR TITLE
add HW_4

### PR DESCRIPTION
**Pull request by dymekkqa**

### CODE REVIEW

File: HW_4/task_2.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+function returnWordWithMaxSymbols(sentence){
+    let array = [];
+    const words = sentence.split(' ');
+    let count = 0;
+    for(let word of words){
+        if(word.length > count){
+            array = []; 
+            array.push(word);
```
Comment (VladimirLQA) [inline HW_4/task_2.js]:
```suggestion
            array = [word]; 
```

File: HW_4/task_2.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        }
+        else if(word.length === count){
+            array.push(word);
+        };        
+    }
+    return array.join(',')
+};
+
+
+function deleteNumbersAreNotEqual(arr) {
```
Comment (VladimirLQA) [inline HW_4/task_2.js]:
```suggestion
function getUniqueValues(arr) {    
```

File: HW_4/task_2.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            array.push(word);
+        };        
+    }
+    return array.join(',')
+};
+
+
+function deleteNumbersAreNotEqual(arr) {    
+    let result = new Set(arr);
+    return Array.from(result);
```
Comment (VladimirLQA) [inline HW_4/task_2.js]: Can do it in one line (with `spread` would be more concise)

File: HW_4/task_3.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    и так пока в тексте не останется двух одинаковых букв стоящих рядом (через пробел и другой знак препинания можно)
+    Пример: aabc => bbc => cc => d
+
+*/
+function reсursionSumNumbers(num){
+    if(num < 10){
+        return num;
+    }
+    let sum = 0;
+    for(let i of String(num).split('')){
```
Comment (VladimirLQA) [inline HW_4/task_3.js]:
```suggestion
    for(const i of String(num)){

```
No need additional .split for a string, `for...of` is used for iterable objects - both arrays and strings are iterables in JS

File: HW_4/task_3.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        sum += Number(i);
+    }
+    return reсursionSumNumbers(sum)
+}
+console.log(reсursionSumNumbers(912345))
+
+
+function recursionEqualSymbols(str) {
```
Comment (VladimirLQA) [inline HW_4/task_3.js]: Solution works correctly, but try to solve it with less lines (not necessary)
As main condition u can use `str[0] !== str[1]` or `str.length > 1`, some .replace, .replaceAll, .slice can come in handy

**Review by VladimirLQA — CHANGES_REQUESTED** [review]

File: HW_4/task_2.js (hunk 3 above)
Comment (dymekkqa) [reply HW_4/task_2.js]: Yes, I found  a solution. Nice, thanks :)

File: HW_4/task_2.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+4. Написать функцию, которая принимает на вход слово и проверяет, является ли это слово палиндромом
+
+*/
+
+function checkCompetitorPizzas(myPizzas){
+    const competitorPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai'];
+    let result = [];
+    for(pizza of competitorPizzas) {
+        if(!myPizzas.includes(pizza)){
```
Comment (danikkrapivnitskiy) [inline HW_4/task_2.js:20]: function doesn't check if the same pizza has different case. 
Try exp ['Peperoni', 'hawai', 'machette', 'caprioni', 'diablo']

Comment (dymekkqa) [reply HW_4/task_2.js:20]: ok, but its a edge case. This task doesnt include Upper or Lower case

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**